File: 3.1P/24HourClock/Clock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _24HourClock
{
    public class Clock
    {
        private Counter _seconds;
        private Counter _minutes;
        private Counter _hours;

        public string Time
        {
            get
            {
                return string.Format("{0:00}:{1:00}:{2:00}", _hours.Ticks, _minutes.Ticks, _seconds.Ticks);
            }
        }
           
        public Clock()
        {
            _seconds = new Counter("seconds");
            _minutes = new Counter("minutes");
            _hours = new Counter("hours");            
        }

        public void Tick()
        {
            _seconds.Increment();
            if (_seconds.Ticks == 60)
            {
                _minutes.Increment();
                _seconds.Reset();
            }
            if (_minutes.Ticks == 60)
            {
                _hours.Increment();
                _minutes.Reset();
            }
            if ( _hours.Ticks == 24)
            {
                _hours.Reset();
            }
        }

        public void Reset()
        {
            _seconds.Reset();
            _minutes.Reset();
            _hours.Reset();
        }

    }
}

File: 11.1P/24HourClock/clock.py
from counter import Counter

class Clock:
    def __init__(self):
        self._seconds = Counter("seconds")
        self._minutes = Counter("minutes")
        self._hours = Counter("hours")
    
    @property
    def time(self):
        return "{:02d}:{:02d}:{:02d}".format(self._hours.ticks, self._minutes.ticks, self._seconds.ticks)
    
    def tick(self):
        self._seconds.increment()
        if self._seconds.ticks == 60:
            self._minutes.increment()
            self._seconds.reset()
            if self._minutes.ticks == 60:
                self._hours.increment()
                self._minutes.reset()
                if self._hours.ticks == 24:
                    self._hours.reset()
    
    def reset(self):
        self._seconds.reset()
        self._minutes.reset()
        self._hours.reset()
        